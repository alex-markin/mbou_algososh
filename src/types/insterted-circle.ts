export type SubCircle = {
  direction: string;
  component: any;
};
