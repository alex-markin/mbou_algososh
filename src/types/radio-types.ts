export enum RadioTypes {
  Selection = "selectionSort",
  Bubble = "bubbleSort",
}
