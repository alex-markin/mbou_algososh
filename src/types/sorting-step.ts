export interface SortingStep {
  currentArray: number[];
  indexA: number;
  indexB: number;
  sortedArray: number[];
}