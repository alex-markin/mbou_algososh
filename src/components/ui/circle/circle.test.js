import { render } from '@testing-library/react';
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';


describe('Circle component', () => {

  const MockComponent = () => <div>Mock Component</div>

  it('renders circle with text', () => {
    const { container } = render(<Circle text="Click" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle without text', () => {
    const { container } = render(<Circle />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with head', () => {
    const { container } = render(<Circle head={'head'} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with head component', () => {
    const { container } = render(<Circle head={<MockComponent />} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with tail', () => {
    const { container } = render(<Circle tail={'tail'} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with tail component', () => {
    const { container } = render(<Circle tail={<MockComponent />} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with index', () => {
    const { container } = render(<Circle index={1} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with isSmall props', () => {
    const { container } = render(<Circle isSmall={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with Default state', () => {
    const { container } = render(<Circle state={ElementStates.Default} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with Changing state', () => {
    const { container } = render(<Circle state={ElementStates.Changing} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders circle with Modified state', () => {
    const { container } = render(<Circle state={ElementStates.Modified} />);
    expect(container.firstChild).toMatchSnapshot();
  });

});

