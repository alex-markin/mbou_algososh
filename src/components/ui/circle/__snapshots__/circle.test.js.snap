// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component renders circle with Changing state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   changing"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders circle with Default state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders circle with Modified state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   modified"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders circle with head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders circle with head component 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      Mock Component
    </div>
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders circle with index 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component renders circle with isSmall props 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders circle with tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle component renders circle with tail component 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      Mock Component
    </div>
  </div>
</div>
`;

exports[`Circle component renders circle with text 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component renders circle without text 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
